src/doc/style-guide/src/README.md

```diff
@@ -99,6 +99,42 @@ fn bar() {}
 fn baz() {}
 ```
 
+### Sorting
+
+In various cases, the default Rust style specifies to sort things. If not
+otherwise specified, such sorting should be "version sorting", which ensures
+that (for instance) `x8` comes before `x16` even though the character `1` comes
+before the character `8`. (If not otherwise specified, version-sorting is
+lexicographical.)
+
+For the purposes of the Rust style, to compare two strings for version-sorting:
+
+- Compare the strings by (Unicode) character as normal, finding the index of
+  the first differing character. (If the two strings do not have the same
+  length, this may be the end of the shorter string.)
+- For both strings, determine the sequence of ASCII digits containing either
+  that character or the character before. (If either string doesn't have such a
```

Comment:
The "either ... or" here seems ambiguous -- which one should the algorithm use?

Reply:
It's not ambiguous, because if *both* that character and the character before are digits, they're part of the same sequence. So there's only ever one possible sequence to select here.

Reply:
Alternative equivalent wording welcome, though.

Reply:
Oh you mean it should look for a sequence of digits backwards from the difference point? Like, if the strings are `x112` and `x113`, the difference is at the last letter, but the extracted sequences would be `112` or `113`? I thought the extracted sequences would be `2` and `3`.

I would say explicitly that if either string has a character at that position that is not an ASCII digit, then we just compare the normal ASCII way. Else we consider (on both sides) the longest sequence of ASCII digits that contains that position (or ends there, in case the position is at the end of the string), and we compare them numerically.

Reply:
Ah no you still want the number treatment for `x9a` vs `x10`, of course.

What about "the longest sequence of ASCII digits that either contains or ends at that index"?